### PR TITLE
SISRP-13260, Delegate Students card on Toolbox - basic wiring

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.berkeley.edu/browse/SISRP-13260

The design has not been applied to `delegate_students.html` but it is working in terms of start / stop delegate-act-as mode. The card pays attention to feature flag.

### CODE REVIEW

File: src/assets/javascripts/angular/controllers/widgets/delegateStudentsController.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var _ = require('lodash');
+var angular = require('angular');
+
+/**
+ * Controller gets student linked to delegate user account
+ */
+angular.module('calcentral.controllers').controller('DelegateStudentsController', function(apiService, adminFactory, delegateFactory, $scope) {
+  angular.extend($scope, {
```
Comment (christianvuerings) [inline src/assets/javascripts/angular/controllers/widgets/delegateStudentsController.js]: @johncrossman any reason we need to do an `extend` here and not just set it?

```
$scope.delegateStudents = {
  isLoading: true
};
```

File: src/assets/javascripts/angular/controllers/widgets/delegateStudentsController.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      isLoading: true
+    }
+  });
+
+  var getStudents = function() {
+    delegateFactory.getStudents().success(function(data) {
+      angular.extend($scope, _.get(data, 'feed'));
+      $scope.delegateStudents.isLoading = false;
+    }).error(function() {
+      $scope.delegate.errorMessage = 'The system failed to get students for current delegate.';
```
Comment (christianvuerings) [inline src/assets/javascripts/angular/controllers/widgets/delegateStudentsController.js]: @johncrossman in which template are we showing this `errorMessage`?

File: src/assets/javascripts/angular/controllers/widgets/delegateStudentsController.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ */
+angular.module('calcentral.controllers').controller('DelegateStudentsController', function(apiService, adminFactory, delegateFactory, $scope) {
+  angular.extend($scope, {
+    delegateStudents: {
+      isLoading: true
+    }
+  });
+
+  var getStudents = function() {
+    delegateFactory.getStudents().success(function(data) {
```
Comment (christianvuerings) [inline src/assets/javascripts/angular/controllers/widgets/delegateStudentsController.js]: @johncrossman probably good to add a `return` here as well (which will allow chaining)

File: src/assets/javascripts/angular/factories/delegateFactory.js [hunk 4]
```diff
@@ -21,6 +25,7 @@ angular.module('calcentral.factories').factory('delegateFactory', function(apiSe
   };
 
   return {
+    getStudents: getStudents,
```
Comment (christianvuerings) [inline src/assets/javascripts/angular/factories/delegateFactory.js]: @johncrossman we usually alphabetize these

File: src/assets/javascripts/angular/controllers/widgets/delegateStudentsController.js (hunk 2 above)
Comment (johncrossman) [inline src/assets/javascripts/angular/controllers/widgets/delegateStudentsController.js]: That piece is not done so I'm going to take this out (for now).

**Comment by johncrossman** [comment]
Making suggested changes and fixing jshint errors.

**Comment by johncrossman** [comment]
Thanks @christianvuerings  Changes made.

**Comment by christianvuerings** [comment]
lgtm - feel free to (self)merge when travis is green

**Pull request merged**